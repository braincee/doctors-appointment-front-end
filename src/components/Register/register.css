.register-section {
  background-image: url("../assets/login.jpg");
  background-size: cover;
  height: 100vh;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
}

.register-container {
  background-color: #b4b2ae3f;
  -webkit-backdrop-filter: blur(2px);
  backdrop-filter: blur(1px);
  padding: 80px;
  border-radius: 10px;
}

.register-title {
  text-align: center;
  color: rgba(129, 124, 116, 0.986);
  margin-bottom: 40px;
}

.register-form {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  gap: 65px;
  margin-bottom: 5px;
}

form input {
  padding: 10px;
  border: 1px hidden;
  border-radius: 5px;
}

.button1:hover {
  color: #000000e5;
  background-color: #555453b4;
  font-family: 'Times New Roman', Times, serif;
}
