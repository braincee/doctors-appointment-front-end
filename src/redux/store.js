import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';




const initialState = {};
const middleware = [thunk];

const store = createStore(initialState, (applyMiddleware(...middleware)));