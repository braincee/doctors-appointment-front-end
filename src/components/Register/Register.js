/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { userRegister } from '../../redux/user/login';
import './register.css';

const Register = () => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [error, setErrors] = useState('');

  const newState = useSelector((state) => state.user);

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleChange = (event) => {
    const input = event.target;
    const newvalue = input.value;
    switch (input.name) {
      case 'name':
        setName(newvalue);
        return name;
      case 'email':
        setEmail(newvalue);
        return email;

      default:
        return 'yes';
    }
  };

  useEffect(() => {
    if (newState.status === 200) {
      navigate('/doctors');
    } else if (newState.status !== 200 && newState.status !== '') {
      if (newState.fetchedData.error === 422) {
        setErrors(newState.fetchedData.errors);
      } else {
        setErrors(newState.fetchedData.errors);
      }
    }
  }, [newState]);

  const handleSubmit = (e) => {
    const form = e.target;
    e.preventDefault();
    dispatch(userRegister(name, email));
    form.reset();
  };

  return (
    <section className="register-section">
      <div className="register-container">
        <p>{error}</p>
        <h2 className="register-title">Sign Up</h2>
        <form onSubmit={handleSubmit} className="register-form">
          <input
            type="text"
            className="form-control"
            name="name"
            id="name"
            value={name}
            onChange={handleChange}
            placeholder="Enter your name"
            required
          />
          <input
            type="text"
            className="form-control"
            name="email"
            id="Email"
            value={email}
            onChange={handleChange}
            placeholder="Email"
            required
          />
          <div className="d-flex">
            <button type="submit" className="button1 btn btn-secondary me-4 fs-4">
              Sign Up
            </button>
            <NavLink to="/login"><button type="button" className="button1 btn btn-secondary me-4 fs-4">Log in</button></NavLink>
          </div>
        </form>
      </div>
    </section>
  );
};

export default Register;
