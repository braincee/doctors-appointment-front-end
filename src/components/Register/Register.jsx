/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { NavLink } from 'react-router-dom';
import './register.css';

const Register = () => (
  <section className="register-section">
    <div className="register-container">
      <h2 className="register-title">Sign Up</h2>
      <form className="register-form">
        <input
          type="text"
          name="email"
          placeholder="Email Address"
          required
        />
        <input
          type="text"
          className="form-control"
          id="Name"
          placeholder="Name"
          required
        />
        <input
          type="text"
          className="form-control"
          id="Username"
          placeholder="Name"
          required
        />
        <div className="d-flex">
          <button type="submit" className="button1 btn btn-secondary me-4 fs-4">
            Sign Up
          </button>
          <NavLink to="/login"><button type="button" className="button1 btn btn-secondary me-4 fs-4">Log in</button></NavLink>
        </div>
      </form>
    </div>
  </section>
);

export default Register;
