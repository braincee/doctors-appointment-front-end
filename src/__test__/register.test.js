import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Register from '../components/Register/Register';
import store from '../redux/configureStore';

describe('Register tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Register />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Register page has sign up button', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Register />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Sign Up')).toBeTruthy();
  });

  it('Sign up page has a log in link', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Register />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Log in')).toBeTruthy();
  });
});
