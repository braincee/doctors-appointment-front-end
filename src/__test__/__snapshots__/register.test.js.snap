// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="register-section"
      >
        <div
          class="register-container"
        >
          <p />
          <h2
            class="register-title"
          >
            Sign Up
          </h2>
          <form
            class="register-form"
          >
            <input
              class="form-control"
              id="name"
              name="name"
              placeholder="Enter your name"
              required=""
              type="text"
              value=""
            />
            <input
              class="form-control"
              id="Email"
              name="email"
              placeholder="Email"
              required=""
              type="text"
              value=""
            />
            <div
              class="d-flex"
            >
              <button
                class="button1 btn btn-secondary me-4 fs-4"
                type="submit"
              >
                Sign Up
              </button>
              <a
                class=""
                href="/login"
              >
                <button
                  class="button1 btn btn-secondary me-4 fs-4"
                  type="button"
                >
                  Log in
                </button>
              </a>
            </div>
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="register-section"
    >
      <div
        class="register-container"
      >
        <p />
        <h2
          class="register-title"
        >
          Sign Up
        </h2>
        <form
          class="register-form"
        >
          <input
            class="form-control"
            id="name"
            name="name"
            placeholder="Enter your name"
            required=""
            type="text"
            value=""
          />
          <input
            class="form-control"
            id="Email"
            name="email"
            placeholder="Email"
            required=""
            type="text"
            value=""
          />
          <div
            class="d-flex"
          >
            <button
              class="button1 btn btn-secondary me-4 fs-4"
              type="submit"
            >
              Sign Up
            </button>
            <a
              class=""
              href="/login"
            >
              <button
                class="button1 btn btn-secondary me-4 fs-4"
                type="button"
              >
                Log in
              </button>
            </a>
          </div>
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
